### PR TITLE
[WIP] Delete maintenance activity by id, List maintenance activity for given asset id

### PR DESCRIPTION
**Pull request by ranbharerp15**
Added apis for 
1) Delete maintenance activity by id
2) list all maintenance activities for given asset id
3) Update maintenance activity by id
4) added authentication middleware

### CODE REVIEW

**Comment by ranbharerp15** [comment]
One question:
For list maintenance activities api, if asset_id is not present in db, should we return 404 error code?

In current implementation, it is returning empty array in response. As assets Apis are not available at the moment, I haven't added above check.

File: contract/asset_maintenance.go [hunk 1]
```diff
@@ -1,9 +1,38 @@
...
-import "time"
+import (
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/vkhichar/asset-management/domain"
+)
+
+const (
+	DATE_FORMAT = "2006-01-02" // yyyy-mm-dd
```
Comment (vkhichar) [inline contract/asset_maintenance.go]: follow camel case for variable/constant names

File: handler/util.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/vkhichar/asset-management/contract"
+	"github.com/vkhichar/asset-management/service"
+)
+
+var (
```
Comment (vkhichar) [inline handler/util.go]: we already have a 'customerrors' package for this kind of errors

File: handler/util.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	ErrMissingToken = errors.New("missing token")
+	ErrInvalidToken = errors.New("invalid or expired token")
+	ErrForbidden    = errors.New("forbidden")
+	ErrBadRequest   = errors.New("bad request")
+)
+
+func ReadTokenFromRequest(r *http.Request) (string, error) {
```
Comment (vkhichar) [inline handler/util.go]: move out of 'utils.go' file and create a separate file. this doesn't seems an utility

Comment (vkhichar) [reply handler/util.go]: Probably write a middleware and add that middleware to router level

File: handler/util.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+func verifyToken(r *http.Request, w http.ResponseWriter, adminApi bool) (*service.Claims, error) {
+	token, err := ReadTokenFromRequest(r)
+
+	if err != nil {
+		fmt.Printf("handler: invalid token: %s", err.Error())
+		return nil, ErrMissingToken
+	}
+
+	claims, err := deps.tokenService.ValidateToken(token)
```
Comment (vkhichar) [inline handler/util.go]: dependencies should be injected. so that unit testing would be easy

**Review by vkhichar — CHANGES_REQUESTED** [review]

**Pull request merged**